### PR TITLE
Extracting api information

### PR DESCRIPTION
**Pull request by keoghrmj**
Successfully connecting to the API and retrieving the relevant information. Please let me know if you want any more features you want me to add.

### CODE REVIEW

File: pipeline/Api_information.csv [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name,email,phone,longitude,latitude,closest_town,ISO_code,plant_name,plant_scientific_name,plant_image_url,temperature,soil_moisture,recording_taken,last_watered
```
Comment (SurinaCS) [inline pipeline/Api_information.csv:1]: does this need to be in the gitignore?

File: pipeline/Extract.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import csv
+import requests
+
+def extract_botanist_information(botanist_info: dict) -> dict:
+    """Extracts the botanist information from the API JSON"""
+    return {"name" : botanist_info.get('name'),
+        "email" : botanist_info.get('email'),
+        "phone" : botanist_info.get('phone')}
+
+def extract_location_information(location_info: list) -> dict:
```
Comment (SurinaCS) [inline pipeline/Extract.py:12]: nice type hints

File: pipeline/Extract.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                "ISO_code" : location_info[3]}
+
+def extract_plant_information(plant_info: dict) -> dict:
+    """Extracts the plant information from the API JSON"""
+    scientific_name = plant_info.get('scientific_name')
+    images = plant_info.get('images')
+
+    if scientific_name is not None:
+        scientific_name = scientific_name[0]
+    else:
```
Comment (SurinaCS) [inline pipeline/Extract.py:26]: are the elses necessary?

**Review by qrafiq1 — APPROVED** [review]
Great clean code

File: pipeline/Extract.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                "plant_name": Plant["name"],
+                "plant_scientific_name": Plant["scientific_name"],
+                "plant_image_url": Plant["original_url"],
+                **Plant_metric
+            }
+
+            collected_data.append(combined_data)
+        else:
+            print(f"Failed with status code: {response.status_code}")
+        print(f"Processed plant ID: {number}")
```
Comment (SurinaCS) [inline pipeline/Extract.py:71]: maybe make it clear which plant id failed?

**Review by SurinaCS — APPROVED** [review]
great work, minor change needed for clarity!

**Pull request merged**